src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -534,13 +534,17 @@ private HttpClient CreateHttpClient()
                     httpClientHandler.Proxy = _httpConnectionOptions.Proxy;
                 }
 
-                // Only access HttpClientHandler.ClientCertificates and HttpClientHandler.CookieContainer
-                // if the user has configured those options
-                // Some variants of Mono do not support client certs or cookies and will throw NotImplementedException
-                if (_httpConnectionOptions.Cookies.Count > 0)
+                try
                 {
+                    // On supported platforms, we need to pass the cookie container to the http client
+                    // so that we can capture any cookies from the negotiate response and give them to WebSockets.
                     httpClientHandler.CookieContainer = _httpConnectionOptions.Cookies;
                 }
+                // Some variants of Mono do not support client certs or cookies and will throw NotImplementedException
+                catch (NotImplementedException) { }
```

Comment:
What about `PlatformNotSupportedException`? Some `HttpClientHandler` throw this instead.

E.g. in .NET 5.0 `HttpClientHandler` setting the `CookieContainer` is forwarded to the [underlying handler](https://github.com/dotnet/runtime/blob/b497143c168c8df00f070e95a4c6fee4092d2ea6/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L71) for which browsers will throw PNSE:
https://github.com/dotnet/runtime/blob/b497143c168c8df00f070e95a4c6fee4092d2ea6/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs#L54-L58

---

There are some `Supports*` properties (which could be used here for other settings) but none for whether cookies are supported:
https://github.com/dotnet/runtime/blob/b497143c168c8df00f070e95a4c6fee4092d2ea6/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs#L126-L128

Reply:
:D That's why we test it before merging. For some reason last night I read PlatformNotSupportedException as inheriting NotImplementedException, obviously that's not right